app/models/service_retire_task.rb

```diff
@@ -0,0 +1,95 @@
+class ServiceRetireTask < MiqRequestTask
```

Comment:
@d-m-u The same as requests can all the tasks shared code be in an intermediate class (MiqRetireRequestTask) and the leaf classes just provide overrides. That will allow us to put all shared code in one place.